ocpmodels/trainers/base_trainer.py

```diff
@@ -403,24 +403,29 @@ def load_checkpoint(self, checkpoint_path):
         self.best_val_metric = checkpoint.get("best_val_metric", None)
         self.primary_metric = checkpoint.get("primary_metric", None)
 
-        # Load model, optimizer, normalizer state dict.
-        # if trained with ddp and want to load in non-ddp, modify keys from
-        # module.module.. -> module..
-        first_key = next(iter(checkpoint["state_dict"]))
-        if (
-            not distutils.initialized() or self.config["noddp"]
-        ) and first_key.split(".")[1] == "module":
-            # No need for OrderedDict since dictionaries are technically ordered
-            # since Python 3.6 and officially ordered since Python 3.7
-            new_dict = {k[7:]: v for k, v in checkpoint["state_dict"].items()}
-            self.model.load_state_dict(new_dict)
-        elif distutils.initialized() and first_key.split(".")[1] != "module":
+        # Match the "module." count in the keys of model and checkpoint state_dict
+        # DataParallel model has 1 "module.",  DistributedDataParallel has 2 "module."
+        # Not using either of the above two would have no "module."
+
+        ckpt_key_count = next(iter(checkpoint["state_dict"])).count("module")
+        mod_key_count = next(iter(self.model.state_dict())).count("module")
+        key_count_diff = mod_key_count - ckpt_key_count
+
+        if key_count_diff > 0:
             new_dict = {
-                f"module.{k}": v for k, v in checkpoint["state_dict"].items()
+                key_count_diff * "module." + k: v
+                for k, v in checkpoint["state_dict"].items()
+            }
+        elif key_count_diff < 0:
+            new_dict = {
+                k[7 * abs(key_count_diff) :]: v
```

Comment:
This logic is fine. Maybe for better readability make it explicit we're removing module? Something like `".".join([k.split('module')[key_count_diff:])`. Or even just len("module.") so it's obvious where 7 comes from.